### PR TITLE
Fix libpng and libfreetype port with wasm exceptions are enabled

### PR DESCRIPTION
**Pull request by sbc100**
Sadly, even though libpng can be compiled with setjmp/longjmp support       
some libraries such as openjpeg assume that setjmp support is included.

Fixes: #19001

### CODE REVIEW

File: tools/ports/freetype.py [hunk 1]
```diff
@@ -8,11 +8,20 @@
 TAG = 'version_1'
 HASH = '0d0b1280ba0501ad0a23cf1daa1f86821c722218b59432734d3087a89acd22aabd5c3e5e1269700dcd41e87073046e906060f167c032eb91a3ac8c5808a02783'
 
+variants = {'freetype-wasm-longjmp': {'SUPPORT_LONGJMP': 'wasm'}}
```
Comment (aheejin) [inline tools/ports/freetype.py:11]: Can we use the existing `SjLjLibrary`? https://github.com/emscripten-core/emscripten/blob/e5e97d2acf9a04731096c25094c908f0298e0724/tools/system_libs.py#L773-L804

Comment (sbc100) [reply tools/ports/freetype.py:11]: Not really, the way ports work and the way system libraries work is currently mostly separate.

File: test/test_other.py [hunk 2]
```diff
@@ -92,6 +92,20 @@ def metafunc(self, wasmfs):
   return metafunc
 
 
+def also_with_wasm_longjmp(f):
+  def metafunc(self, wasmfs):
```
Comment (aheejin) [inline test/test_other.py:96]: Why `wasmfs`? Is this related to WasmFS?

File: test/test_other.py [hunk 3]
```diff
@@ -92,6 +92,20 @@ def metafunc(self, wasmfs):
   return metafunc
 
 
+def also_with_wasm_longjmp(f):
```
Comment (aheejin) [inline test/test_other.py:95]: Can we move https://github.com/emscripten-core/emscripten/blob/e5e97d2acf9a04731096c25094c908f0298e0724/test/test_core.py#L127-L149 to `common.py` and use this instead?

File: tools/ports/freetype.py (hunk 1 above)
Comment (aheejin) [reply tools/ports/freetype.py:11]: I see, then how about naming the library `-wasm-sjlj` to be consistent with the naming in `system_libs.py`? https://github.com/emscripten-core/emscripten/blob/e5e97d2acf9a04731096c25094c908f0298e0724/tools/system_libs.py#L793-L794

File: tools/ports/freetype.py [hunk 4]
```diff
@@ -8,11 +8,20 @@
...
+
 
 def needed(settings):
   return settings.USE_FREETYPE
 
 
+def get_lib_name(settings):
+  if settings.SUPPORT_LONGJMP == 'wasm':
+    return 'libfreetype-wasm-eh.a'
```
Comment (aheejin) [inline tools/ports/freetype.py:20]:
```suggestion
    return 'libfreetype-wasm-sjlj.a'
```
How about this to be consistent with https://github.com/emscripten-core/emscripten/blob/e5e97d2acf9a04731096c25094c908f0298e0724/tools/system_libs.py#L793-L794?

File: test/test_other.py (hunk 2 above)
Comment (sbc100) [reply test/test_other.py:96]: copy pasta. fixed.

File: tools/ports/freetype.py (hunk 4 above)
Comment (sbc100) [reply tools/ports/freetype.py:20]: Done

File: test/test_other.py (hunk 3 above)
Comment (sbc100) [reply test/test_other.py:95]: I did try that but it didn't work.   I think it was because this new test is specifically setjmp/longjmp and not about exception handling.  Indeed the test for this, and the libraries themselves are not written in C++ but in C.

I can't remember the exact failure now.. but I think this way is more precise anyway.

Comment (sbc100) [reply test/test_other.py]: Oh sorry I was thinking of `with_both_eh_sjlj`.  

Done!

**Review by aheejin — APPROVED** [review]
Thanks!
